src/classes/ALLO_PaymentSync_TDTM.cls

```diff
@@ -50,6 +50,12 @@ public class ALLO_PaymentSync_TDTM extends TDTM_Runnable {
     public override DmlWrapper run(List<SObject> newList, List<SObject> oldList,
         TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
 
+        Allocations_Settings__c allocationsSettings = UTIL_CustomSettingsFacade.getAllocationsSettings();
+
+        if (allocationsSettings.Payment_Allocation_Two_Way_Sync_Enabled__c == false && !Test.isRunningTest()) {
```

Comment:
Please change this to 

if (!allocationSettings.Payment_Allocation_Two_Way_Sync_Enabled__c && !Test.isRunningTest()) {

this will cover it better because !true covers null as well as false...

Reply:
Please also add a comment indicating that the check for whether tests are running is strictly to cover Abacus until we can refactor the required tests so that they will either explicitly set two way sync enabled to true or are modified to no longer rely on the execution of 2 way sync.  Something like:

// Interim: The exclusion to this for tests is so that existing unit tests in external packages will not fail because they are expecting this to be present.  Once the Abacus tests are refactored and the Payment Allocations Two Way Sync tests are updated, this part of the condition will be removed.

A backlog item for that work item should be logged.

Reply:
This should be after the Payment Allocations Enabled check, not before it.  If Payment Allocations is not Enabled, this setting should not matter at all.  I know it doesn't matter (much) really, but it makes me feel better because it reflects the proper order the settings are/should be applied.

Reply:
Additionally, an item for removing the test shortcircuit, and making sure NPSP Allocations Test all run without it, should also be added.  That item should also explicitly add tests that confirm 2 way sync does/does not execute according to the schedule, since this work item cannot do so.

Reply:
> Please change this to
> 
> if (!allocationSettings.Payment_Allocation_Two_Way_Sync_Enabled__c && !Test.isRunningTest()) {
> 
> this will cover it better because !true covers null as well as false...

Using !someBoolVar will throw a null pointer if the boolean is actually null. Using someBoolVar == false actually covers nulls and true/false.